src/SFML/Graphics/Text.cpp

```diff
@@ -568,4 +702,38 @@ void Text::ensureGeometryUpdate() const
     m_bounds.height = maxY - minY;
 }
 
+
+////////////////////////////////////////////////////////////
+void Text::updateUnderline(VertexArray& arr, const Color& color) const
+{
+    bool isUnderlined    = m_style & Underlined;
+    bool isStrikeThrough = m_style & StrikeThrough;
+
+    //In case of newlines, find them and change lines
+    unsigned int  lineCount = 0;
+    std::uint32_t prevChar  = 0;
+    for (unsigned int i = 0; i < m_string.getSize(); ++i)
+    {
+        std::uint32_t curChar = m_string[i];
+
+        if (isUnderlined && (curChar == U'\n' && prevChar != U'\n'))
+        {
+            for (unsigned int j = 0; j < 6; j++)
```

Comment:
What's the significance of this `6`?

Reply:
SFML uses triangles to render quads (quads aren't available on OpenGLES), therefore the number of vertices per character is 6 instead of 4.
The append() function gets called 6 times for each vertex.
![image](https://user-images.githubusercontent.com/54750550/213581755-e0ec87ab-8409-4e72-a6bd-fc05af2800d0.png)